monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -1,80 +1,152 @@
     };
     // set schema from server
     this.state = {
       schema: {},
       configuration: {},
+      attackConfig: {},
       lastAction: 'none',
       sections: [],
-      selectedSection: 'basic',
+      selectedSection: 'attack',
       allMonkeysAreDead: true,
       PBAwinFile: [],
-      PBAlinuxFile: []
+      PBAlinuxFile: [],
+      showAttackAlert: false
     };
   }
 
-  componentDidMount() {
-    this.authFetch('/api/configuration/island')
-      .then(res => res.json())
-      .then(res => {
+  setInitialConfig(config) {
+    this.initialConfig = JSON.parse(JSON.stringify(config));
+  }
+
+  setInitialAttackConfig(attackConfig) {
+    this.initialAttackConfig = JSON.parse(JSON.stringify(attackConfig));
+  }
+
+  componentDidMount = () => {
+    let urls = [CONFIG_URL, ATTACK_URL];
+    Promise.all(urls.map(url => this.authFetch(url).then(res => res.json())))
+      .then(data => {
         let sections = [];
+        let attackConfig = data[1];
+        let monkeyConfig = data[0];
+        this.setInitialConfig(monkeyConfig.configuration);
+        this.setInitialAttackConfig(attackConfig.configuration);
         for (let sectionKey of this.sectionsOrder) {
-          sections.push({key: sectionKey, title: res.schema.properties[sectionKey].title});
+          if (sectionKey === 'attack') {sections.push({key:sectionKey, title: "ATT&CK"})}
+          else {sections.push({key: sectionKey, title: monkeyConfig.schema.properties[sectionKey].title});}
         }
         this.setState({
-          schema: res.schema,
-          configuration: res.configuration,
+          schema: monkeyConfig.schema,
+          configuration: monkeyConfig.configuration,
+          attackConfig: attackConfig.configuration,
           sections: sections,
-          selectedSection: 'basic'
+          selectedSection: 'attack'
         })
       });
     this.updateMonkeysRunning();
-  }
+  };
 
-  onSubmit = ({formData}) => {
-    this.currentFormData = formData;
+  updateConfig = () => {
+    this.authFetch(CONFIG_URL)
+    .then(res => res.json())
+    .then(data => {
+      this.setInitialConfig(data.configuration);
+      this.setState({configuration: data.configuration})
+    })
+  };
+
+  onSubmit = () => {
+    if (this.state.selectedSection === 'attack'){
+      this.matrixSubmit()
+    } else {
+      this.configSubmit()
+    }
+  };
+
+  matrixSubmit = () => {
+    // Submit attack matrix
+    this.authFetch(ATTACK_URL,
+      {
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(this.state.attackConfig)
+      })
+      .then(res => {
+        if (!res.ok)
+        {
+          throw Error()
+        }
+        return res;
+      })
+      .then(() => {this.setInitialAttackConfig(this.state.attackConfig);
+                   this.setState({lastAction: 'saved'})})
+      .then(this.updateConfig())
+      .catch(error => {
+        console.log('bad attack configuration');
```

Comment:
I don't want the console to become a debug log. There are several good solutions for this, but we haven't implemented any for the project so far.
Let's remove it for now. Start a discussion on `#monkey_dev` regarding logging for debugging on frontend.
I think we should check whether we're on development or production somehow (there are several ways I can think about), and print to console log (or somewhere else) only if on development